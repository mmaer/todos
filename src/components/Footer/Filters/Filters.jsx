import React, { memo } from "react";
import classnames from "classnames";

import { FILTERS } from "../../../constants";

import "./Filters.scss";

const Filters = memo(({ setFilter, filter }) => (
  <ul className="filters">
    {FILTERS.map((link, index) => (
      <li key={index} className="filters__filter">
        <a
          href={`/${link}`}
          className={classnames("filters__link", {
            "filters__link--selected": filter === link
          })}
          onClick={() => setFilter(link)}
        >
          {link[0].toUpperCase() + link.slice(1)}
        </a>
      </li>
    ))}
  </ul>
));

export default Filters;
