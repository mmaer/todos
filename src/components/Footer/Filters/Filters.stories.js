import React from "react";
import { MemoryRouter } from "react-router-dom";

import Filters from ".";

export default {
  title: "Filters",
};

export const filters = () => (
  <MemoryRouter>
    <Filters />
  </MemoryRouter>
);
