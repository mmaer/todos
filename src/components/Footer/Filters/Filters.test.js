import React from "react";
import { shallow } from "enzyme";

import Filters from "./";

describe("Filters component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(<Filters />);
    expect(wrapper).toMatchSnapshot();
  });
});
