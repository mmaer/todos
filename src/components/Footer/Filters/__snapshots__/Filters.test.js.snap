// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters component Should match snapshot 1`] = `
<ul
  className="filters"
>
  <li
    className="filters__filter"
    key="filter-0"
  >
    <NavLink
      activeClassName="filters__link--selected"
      className="filters__link"
      to="/all"
    >
      All
    </NavLink>
  </li>
  <li
    className="filters__filter"
    key="filter-1"
  >
    <NavLink
      activeClassName="filters__link--selected"
      className="filters__link"
      to="/active"
    >
      Active
    </NavLink>
  </li>
  <li
    className="filters__filter"
    key="filter-2"
  >
    <NavLink
      activeClassName="filters__link--selected"
      className="filters__link"
      to="/completed"
    >
      Completed
    </NavLink>
  </li>
</ul>
`;
